### PR TITLE
Gather amp-version from apicast-staging trigger

### PR DESCRIPTION
**Pull request by pehala**
This should make sure the version is correct even in upgraded environments, as image triggers are present and updated in both template and operator upgrade/install.

### CODE REVIEW

File: testsuite/dynaconf_loader.py [hunk 1]
```diff
@@ -53,17 +53,14 @@ def _guess_version(ocp):
...
-    lookup = _docker_image(ocp, "amp-system")
+    config = ocp.do_action("get", ["dc/apicast-staging", "-o", "yaml"], parse_output=True)
+    lookup = [trigger.imageChangeParams["from"]["name"].split(":", 1)[1]
+              for trigger in config.model.spec.triggers if trigger.type == "ImageChange"][0]
 
     try:
-        version = lookup["name"]
-        Version(version)
+        Version(lookup)
     except (KeyError, InvalidVersion):
```
Comment (mganisin) [inline testsuite/dynaconf_loader.py:62]: if original code expected KeyError this should probably handle IndexError and try should cover one line more

File: testsuite/dynaconf_loader.py [hunk 2]
```diff
@@ -53,17 +52,13 @@ def _guess_version(ocp):
     """Attempt to determine version from amp-system imagestream"""
 
     version = None
-    lookup = _docker_image(ocp, "amp-system")
```
Comment (mganisin) [inline testsuite/dynaconf_loader.py:56]: `_docker_image` is now unused

File: testsuite/dynaconf_loader.py [hunk 3]
```diff
@@ -53,17 +52,13 @@ def _guess_version(ocp):
...
-
     try:
-        version = lookup["name"]
+        config = ocp.do_action("get", ["dc/apicast-staging", "-o", "yaml"], parse_output=True)
```
Comment (mganisin) [inline testsuite/dynaconf_loader.py:56]: original code did derivate version from `amp-system`... ???

**Review by mganisin — CHANGES_REQUESTED** [review]

File: testsuite/dynaconf_loader.py (hunk 3 above)
Comment (pehala) [reply testsuite/dynaconf_loader.py:56]: amp-system can have a different image stream tag if used with Oracle so I changed it to apicast instead.

Comment (mganisin) [reply testsuite/dynaconf_loader.py:56]: apicast-staging can have different image stream tag if custom-policy is injected (yeah, less probable). apicast-production?

Comment (pehala) [reply testsuite/dynaconf_loader.py:56]: I don't think anyone is (and should be) manipulating apicast-staging deployment, however I will change it to apicast-production just to be safe

Comment (pehala) [reply testsuite/dynaconf_loader.py]: Changed

**Review by JaurbanRH — APPROVED** [review]

**Review by mganisin — APPROVED** [review]

**Pull request merged**